src/cascadia/TerminalSettingsModel/ActionMapSerialization.cpp

```diff
@@ -59,32 +59,52 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
+            Control::KeyChord keys{ nullptr };
+            if (withKeybindings && jsonBlock.isMember(JsonKey(KeysKey)))
+            {
+                const auto keysJson{ jsonBlock[JsonKey(KeysKey)] };
+                if (keysJson.isArray() && keysJson.size() > 1)
+                {
+                    warnings.push_back(SettingsLoadWarnings::TooManyKeysForChord);
+                }
+                else
+                {
+                    JsonUtils::GetValueForKey(jsonBlock, KeysKey, keys);
+                }
+            }
+
+            // Now check if this is a command block
             if (jsonBlock.isMember(JsonKey(CommandsKey)) || jsonBlock.isMember(JsonKey(ActionKey)))
             {
-                AddAction(*Command::FromJson(jsonBlock, warnings, origin, withKeybindings));
+                AddAction(*Command::FromJson(jsonBlock, warnings, origin), keys);
 
-                // for non-nested non-iterable commands,
-                // check if this is a legacy-style command block so we can inform the loader that fixups are needed
-                if (jsonBlock.isMember(JsonKey(ActionKey)) && !jsonBlock.isMember(JsonKey(IterateOnKey)))
+                if (jsonBlock.isMember(JsonKey(KeysKey)))
                 {
-                    if (jsonBlock.isMember(JsonKey(KeysKey)))
-                    {
-                        // there are keys in this command block - its the legacy style
-                        _fixUpsAppliedDuringLoad = true;
-                    }
-
-                    if (origin == OriginTag::User && !jsonBlock.isMember(JsonKey(IDKey)))
-                    {
-                        // there's no ID in this command block - we will generate one for the user
-                        // inform the loader that the ID needs to be written into the json
-                        _fixUpsAppliedDuringLoad = true;
-                    }
+                    // there are keys in this command block meaning this is the legacy style -
+                    // inform the loader that fixups are needed
+                    _fixUpsAppliedDuringLoad = true;
+                }
+
+                if (jsonBlock.isMember(JsonKey(ActionKey)) && !jsonBlock.isMember(JsonKey(IterateOnKey)) && origin == OriginTag::User && !jsonBlock.isMember(JsonKey(IDKey)))
+                {
+                    // for non-nested non-iterable commands,
+                    // if there's no ID in the command block we will generate one for the user -
+                    // inform the loader that the ID needs to be written into the json
+                    _fixUpsAppliedDuringLoad = true;
                 }
             }
-            else
+            else if (keys)
             {
                 // this is not a command block, so it is a keybinding block
-                _AddKeyBindingHelper(jsonBlock, warnings);
+
+                // if the "id" field doesn't exist in the json, then idJson will be an empty string which is fine

```

Comment:
huh so I guess `{ "keys": "ctrl+shift+c" }` is enough to unbind a key?

Reply:
Yep!